wallet/src/de/schildbach/wallet/service/WalletTransactionMetadataProvider.kt

```diff
@@ -34,18 +33,18 @@ import javax.inject.Inject
 
 class WalletTransactionMetadataProvider @Inject constructor(
     private val transactionMetadataDao: TransactionMetadataDao,
-    private val walletApplication: WalletApplication
+    private val walletData: WalletDataProvider
```

Comment:
Getting the wallet from `WalletDataProvider` here to keep abstracting it from the `WalletApplication`